core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -97,6 +97,24 @@ public static boolean isIngestNode(Settings settings) {
     private final Version version;
     private final Set<Role> roles;
 
+
+    /**
+     * Creates a new {@link DiscoveryNode}
+     * <p>
+     * <b>Note:</b> if the version of the node is unknown {@link Version#minimumCompatibilityVersion()} should be used for the current
+     * version. it corresponds to the minimum version this elasticsearch version can communicate with. If a higher version is used
+     * the node might not be able to communicate with the remove node. After initial handshakes node versions will be discovered
+     * and updated.
+     * </p>
+     *
+     * @param id               the nodes unique (persistent) node id. This constructor will auto generate a random ephemeral id.
+     * @param address          the nodes transport address
+     * @param version          the version of the node
+     */
+    public DiscoveryNode(final String id, TransportAddress address, Version version) {
```

Comment:
There have been many times that I've wanted this constructor and this time I've finally added it (the verbosity of any of the other constructors just gets in the way when writing tests since we often passing empty for the attributes and roles). I'm happy to move it to a convenience method in the test sources, but I would prefer to keep it.